### PR TITLE
Issue #SB-17236 feat : Handling  storage space running out 

### PR DESCRIPTION
**Pull request by santoshilimi**

### CODE REVIEW

File: src/app/client/projects/desktop/src/app/modules/offline/components/content-manager/content-manager.component.html [hunk 1]
```diff
@@ -147,3 +147,6 @@
     </div>
   </sui-accordion-panel>
 </sui-accordion>
+
+
+<app-nospace-content-list *ngIf="listToShow['length']" [contentlistToShow]="listToShow" (dismissed)="closeModal($event)"> </app-nospace-content-list>
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/content-manager/content-manager.component.html]: listToShow?.length

Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/content-manager/content-manager.component.html]: listToShow.length

File: src/app/client/projects/desktop/src/app/modules/offline/components/content-manager/content-manager.component.ts [hunk 2]
```diff
@@ -25,6 +25,10 @@ export class ContentManagerComponent implements OnInit, OnDestroy {
   apiCallTimer = timer(1000, 3000).pipe(filter(data => !data || (this.callContentList)));
   apiCallSubject = new Subject();
   completedCount: number;
+  noSpaceContentList: any;
+  previousList: any;
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/content-manager/content-manager.component.ts]: only two property is required.
remove noSpaceContentList
and rename 
listToShow  > unHandledFailedList 
previousList  > handledFailedList

File: src/app/client/projects/desktop/src/app/modules/offline/components/content-manager/content-manager.component.ts [hunk 3]
```diff
@@ -76,9 +80,21 @@ export class ContentManagerComponent implements OnInit, OnDestroy {
           this.contentResponse = _.filter(apiResponse, (o) => {
             return o.status !== 'canceled';
           });
+          this.getNoSpaceContentList(apiResponse);
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/content-manager/content-manager.component.ts]: handleInsufficentMemoryError

File: src/app/client/projects/desktop/src/app/modules/offline/components/content-manager/content-manager.component.ts [hunk 4]
```diff
@@ -76,9 +80,21 @@ export class ContentManagerComponent implements OnInit, OnDestroy {
...
         });
   }
-
+  getNoSpaceContentList(allContentList) {
+    this.noSpaceContentList = _.filter(allContentList, (content) =>
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/content-manager/content-manager.component.ts]: noSpaceContentList not required. do forEach and to listToShow, if it's not in previously shown and not in listToShow also

File: src/app/client/projects/desktop/src/app/modules/offline/components/content-manager/content-manager.component.ts [hunk 5]
```diff
@@ -76,9 +80,21 @@ export class ContentManagerComponent implements OnInit, OnDestroy {
...
+    content.failedCode === 'LOW_DISK_SPACE' && content.status === 'failed');
+    this.listToShow =  _.differenceBy(this.noSpaceContentList , this.previousList, 'identifier');
+  }
+  removeFromPreviousList(id) {
+    this.previousList = _.filter(this.previousList, (content) => content.id !== id);
+  }
+  closeModal() {
+    this.previousList = this.noSpaceContentList;
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/content-manager/content-manager.component.ts]: this.previousList.push(this. listToShow)

File: src/app/client/projects/desktop/src/app/modules/offline/components/content-manager/content-manager.component.ts [hunk 6]
```diff
@@ -25,6 +25,10 @@ export class ContentManagerComponent implements OnInit, OnDestroy {
   apiCallTimer = timer(1000, 3000).pipe(filter(data => !data || (this.callContentList)));
   apiCallSubject = new Subject();
   completedCount: number;
+  noSpaceContentList: any;
+  previousList: any;
+  listToShow = [];
+  @Output() contentlistToShow = new EventEmitter<any>();
```
Comment (anoophm) [inline src/app/client/projects/desktop/src/app/modules/offline/components/content-manager/content-manager.component.ts]: why we need this

File: src/app/resourcebundles/data/consumption/en.properties [hunk 7]
```diff
@@ -974,6 +974,9 @@ messages.stmsg.desktop.noContentMessage = Content is coming soon
 messages.stmsg.desktop.deleteErrorMessage = Unable to delete content.Please tryagain..
 messages.stmsg.desktop.deleteSuccessMessage = Content Deleted successfully
 frmelmnts.lbl.desktop.creditsAndLicenceInfo = Credits And Licence Info
+frmelmnts.lbl.desktop.btn.OK = OK
```
Comment (anoophm) [inline src/app/resourcebundles/data/consumption/en.properties]: dont create new property for ok, resue existing.

**Comment by santoshilimi** [comment]
PR raised to 2.8.0 so i am closing this PR

**Pull request closed without merging**